_pytest/python.py

```diff
@@ -49,7 +49,10 @@ def _has_positional_arg(func):
 
 
 def filter_traceback(entry):
-    return entry.path != cutdir1 and not entry.path.relto(cutdir2)
+    # ensure entry.path is always a py.path.local object
+    # see https://bitbucket.org/pytest-dev/py/issues/71
+    path = py.path.local(entry.path)
```

Comment:
OK, I will implement this change later then.

That aside, how about adding a `entry.frame.code.is_generated()` to abstract this away from clients and not having them hard-coding the `"<...>"`-detection logic?